### PR TITLE
Add support for GitHub Apps (formerly called Integrations)

### PR DESCRIPTION
**Pull request by Keno**

### CODE REVIEW

File: src/utils/auth.jl [hunk 1]
```diff
@@ -10,6 +10,37 @@ end
...
+        "iss" => app_id
+    ))))
+    entropy = MbedTLS.Entropy()
+    rng = MbedTLS.CtrDrbg()
+    MbedTLS.seed!(rng, entropy)
+    key = MbedTLS.parse_keyfile(priv_key)
+    signature = base64_to_base64url(base64encode(MbedTLS.sign(key, MbedTLS.MD_SHA256,
+        MbedTLS.digest(MbedTLS.MD_SHA256, string(algo,'.',data)), rng)))
+    JWTAuth(string(algo,'.',data,'.',signature))
+end
```
Comment (jrevels) [inline src/utils/auth.jl]: Would it be reasonable to add a simple test for this with a dummy key?

File: src/apps/installations.jl [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+end
+
+namefield(i::Installation) = i.id
+
+Installation(data::Dict) = json2github(Installation, data)
+Installation(id::Int) = Installation(Dict("id" => id))
+
+function create_access_token(i::Installation, auth::JWTAuth; headers = Dict(), options...)
+    headers["Accept"] = "application/vnd.github.machine-man-preview+json"
+    @show i.id
```
Comment (KristofferC) [inline src/apps/installations.jl]: Should be removed?

**Comment by Keno** [comment]
Updated to add a simple test and remove the debug statement.

**Comment by jrevels** [comment]
LGTM, thanks!

**Pull request merged**